torchsynth/module.py

```diff
@@ -505,43 +486,24 @@ class VCA(SynthModule):
     Voltage controlled amplifier.
     """
 
-    def _forward(self, control_in: Signal, audio_in: Signal) -> Signal:
-        assert (control_in >= 0).all() and (control_in <= 1).all()
-
-        # Should VCA be responsible for this?
-        if (audio_in <= -1).any() or (audio_in >= 1).any():
-            util.normalize(audio_in)
-
-        audio_in = util.fix_length2D(audio_in, control_in.num_samples)
-        return control_in * audio_in
+    def _forward(self, audio_in: Signal, control_in: Signal) -> Signal:
+        return audio_in * control_in
 
 
 class Noise(SynthModule):
     """
-    Adds noise to a signal
+    Generates white noise that is the same length as the buffer
     """
 
-    parameter_ranges: List[ModuleParameterRange] = [
-        ModuleParameterRange(
-            0.0,
-            1.0,
-            curve=0.05,
-            name="ratio",
-            description="mix ratio between the incoming signal and the produced noise, "
-            + "1 is all noise",
-        )
-    ]
-
-    def _forward(self, audio_in: Signal) -> Signal:
-        noise = self.noise_of_length(audio_in)
-        return util.crossfade2D(audio_in, noise, self.p("ratio"))
-
-    def noise_of_length(self, audio_in: Signal) -> Signal:
+    def _forward(self, device: Optional[torch.device] = None) -> Signal:
```

Comment:
Get rid of device and just use self.device

Reply:
device isn't an attribute of nn.Module, just the lightning stuff and SynthModule isn't a lightning module.